### PR TITLE
Extract low level loop to helper function

### PR DESCRIPTION
**Pull request by wallclockbuilder**
Sorry. The sudden change in abstraction, from readable high to detailed low level inner loop was jarring. Couldn't help it. I refactored the out of place low level inner loop into a helper function. 7 lines now replaced with 1 thats readable and does the same thing.

### CODE REVIEW

File: goconvey.go [hunk 1]
```diff
@@ -80,18 +80,29 @@ func main() {
...
+func extractPackages(folderList messaging.Folders) []*contract.Package {
+	packageList := []*contract.Package{}
+	for _, folder := range folderList {
+		hasImportCycle := testFilesImportTheirOwnPackage(folder.Path)
+		packageList = append(packageList, contract.NewPackage(folder, hasImportCycle))
+	}
+	return packageList
+}
+
+func extractRoot(folders messaging.Folders) string {
```
Comment (mdwhatcott) [inline goconvey.go]: Hmm, this function could be modified to simply grab the last folder's root... I think that would be better than looping over the folders twice (once in this function and once in the `extractPackage` function.

Comment (wallclockbuilder) [inline goconvey.go]: The `folderList` is a map, its values can only be accessed by key. I don't think you can get at the values by index.

Comment (wallclockbuilder) [inline goconvey.go]: Double loop fixed.

Used the `path` value of  the first package `packageList[0]` as key to the `folderList`.

**Comment by mdwhatcott** [comment]
Thanks!

**Pull request merged**